libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -1712,6 +1714,7 @@ Compass::save_offsets(uint8_t i)
     if (id < COMPASS_MAX_INSTANCES) {
         _state[id].offset.save();  // save offsets
         _state[id].dev_id.set_and_save(_state[id].detected_dev_id);
+        _state[id].dev_id.set_default(_state[id].detected_dev_id);
```

Comment:
As above, looks like we only sometimes save the offsets, after a calibration of some sort. So there is no point. On in normal operation you will not get this value.

Reply:
and as above, it is useful as when we calibrate we set the same ID, so again this is effective in hiding params set by ardupilot as opposed to being set by the user
the param changes seen in the GCS UI should be the things the user has set

Reply:
I don't think we should try and combine 'not default' with 'changed by user'. Its also useful to know if something has been changed by a calibration. 

Maybe we need a new piece of metadata to mark "calibration" type params, such as the offsets and other things that should not normally be set manually. We can then have a second tick box in the GCS to hide calibration values.

Reply:
the devids are not changed by calibration, they are the sensors you have in the flight controller. Showing them as "changed" is just confusing

Reply:
Your probably right that there not changed in the calibration (I have not checked every case). In which case we don't need the existing `set_and_save` at all (and thus not the new `set_default`).